ray-operator/test/e2e/rayjob_cluster_selector_test.go

```diff
@@ -0,0 +1,278 @@
+package e2e
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	"github.com/ray-project/kuberay/ray-operator/controllers/ray/common"
+	. "github.com/ray-project/kuberay/ray-operator/test/support"
+)
+
+func TestRayJobWithClusterSelector(t *testing.T) {
+	test := With(t)
+
+	// Create a namespace
+	namespace := test.NewTestNamespace()
+
+	// Job scripts
+	jobs := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: corev1.SchemeGroupVersion.String(),
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "jobs",
+			Namespace: namespace.Name,
+		},
+		BinaryData: map[string][]byte{
+			"counter.py": ReadFile(test, "counter.py"),
+			"fail.py":    ReadFile(test, "fail.py"),
+		},
+		Immutable: Ptr(true),
+	}
+	jobs, err := test.Client().Core().CoreV1().ConfigMaps(namespace.Name).Create(test.Ctx(), jobs, metav1.CreateOptions{})
+	test.Expect(err).NotTo(HaveOccurred())
+	test.T().Logf("Created ConfigMap %s/%s successfully", jobs.Namespace, jobs.Name)
+
+	// RayCluster
+	rayCluster := &rayv1.RayCluster{
+		TypeMeta: metav1.TypeMeta{
```

Comment:
Can we reuse definitions from the unit tests? We seem to repeat these definitions all over the place.